test/SlowTests/Issues/RavenDB_20416.cs

```diff
@@ -167,37 +204,120 @@ private async Task CheckSubscriptionFetchAfterUpdate(DocumentStore store1, IDocu
             {
-                string cvs = null;
-                using (db2.ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
-                using (ctx.OpenReadTransaction())
+                var shards1 = Sharding.GetShardsDocumentDatabaseInstancesFor(store1);
+                Assert.NotNull(shards1);
+                var dbIds1 = await shards1.Select(x => x.DbBase64Id).ToListAsync();
+
+                var shards2 = Sharding.GetShardsDocumentDatabaseInstancesFor(store2);
+                Assert.NotNull(shards2);
+                var dbIds2 = await shards2.Select(x => x.DbBase64Id).ToListAsync();
+                Dictionary<string, string> cvs2 = new Dictionary<string, string>();
+                var count2 = 0l;
+                var count1 = 0l;
+                await foreach (var db in shards2)
                 {
-                    var connectionState = db2.SubscriptionStorage.GetSubscriptionConnectionsState(ctx, state.SubscriptionName);
-                    if (connectionState != null)
+                    var newCv = cvs[db.Name];
+                    var cv = WaitForValue(() =>
+                    {
+                        using (db.ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
+                        using (ctx.OpenReadTransaction())
+                        {
+                            var connectionState = db.SubscriptionStorage.GetSubscriptionConnectionsState(ctx, state.SubscriptionName);
+                            if (connectionState != null)
+                            {
+                                var connection = connectionState.GetConnections().FirstOrDefault();
+                                if (connection != null)
+                                {
+                                    var subsState = connection.SubscriptionState;
+                                    if (subsState.ShardingState != null)
+                                    {
+                                        if (subsState.ShardingState.ChangeVectorForNextBatchStartingPointPerShard.TryGetValue(db.Name, out string cv))
+                                        {
+                                            return cv;
+                                        }
+
+                                        return null;
+                                    }
+
+                                    return null;
+                                }
+
+                                return null;
+                            }
+
+                            return null;
+                        }
+                    }, newCv, interval: 333);
+                    Assert.Equal(newCv, cv);
+
+                    DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal,
+                        "enable the commented code after RavenDB-20454 is done");
+
+                    /*var cvList = cv.ToChangeVectorList();
```

Comment:
clean it up please

Reply:
this needs to be uncommented after  RavenDB-20454 is done